### ISSUE TITLE
OAuth support

### ISSUE THREAD

**Issue by Abbe98** [MER-C/wiki-java#153]
OAuth support would be very valuable, especially for users with Two Factor Authentication enabled.

**Comment by MER-C** [comment]
I agree, but this is currently not on my radar. I need to kill off all the non-API using methods first.

**Comment by MER-C** [comment]
The last method using base was rewritten in 18aaaf90d so this can now proceed.

Instructions: https://www.mediawiki.org/wiki/OAuth/Owner-only_consumers (the more complicated flow can be left later). 

The OAuth signature needs to be computed at the last possible moment, immediately before connection.connect() in makeHTTPrequest. OAuth is a MediaWiki extension, so it doesn't belong in Wiki.java. That makes the solution ugly. I'd recommend using bot passwords for now.

**Comment by MER-C** [comment]
All HTTP GET parameters are now in the same format as of 56a923ff .

**Comment by Nirvanchik** [comment]
> OAuth is a MediaWiki extension, so it doesn't belong in Wiki.java. That makes the solution ugly.

That's sad.
